docs/components/index.md

```diff
@@ -0,0 +1,360 @@
+
+## Razor directives
+
+A Razor file specifies [Razor directives](https://docs.microsoft.com/aspnet/core/mvc/views/razor#directives) at the top of the file. Razor directives supported by Blazor are shown in the table. `@model`, `@section`, and Tag Helper directives aren't supported at this time.
+
+| Directive | Description |
+| --------- | ----------- |
+| [@functions](https://docs.microsoft.com/aspnet/core/mvc/views/razor#functions) | Adds a C# code block to a component. |
+| `@implements` | Implements an interface for the generated component class. Often used to create a component layout by implementing `ILayoutComponent`. |
+| [@inherits](https://docs.microsoft.com/aspnet/core/mvc/views/razor#inherits) | Provides full control of the class that the component inherits. |
+| [@inject](https://docs.microsoft.com/aspnet/core/mvc/views/razor#inject) | Enables service injection from the [service container](https://docs.microsoft.com/aspnet/core/fundamentals/dependency-injection). For more information, see [Dependency injection into views](https://docs.microsoft.com/aspnet/core/mvc/views/dependency-injection). |
+| `@layout` | Specifies a layout component. Layout components are used to avoid code duplication and inconsistency. |
+| [@page](https://docs.microsoft.com/aspnet/core/mvc/razor-pages#razor-pages) | Specifies that the component should handle requests directly. The `@page` directive can be specified with a route and optional parameters. Unlike Razor Pages, the `@page` directive doesn't need to be the first directive at the top of the file. |
+| [@using](https://docs.microsoft.com/aspnet/core/mvc/views/razor#using) | Adds the C# `using` directive to the generated component class. |
+
+## Component classes
+
+Blazor's Razor files (*\*.cshtml*) mix HTML markup and C# processing code in the same file, as shown in the sample `HeadingComponent` above. Blazor doesn't currently support `partial` component classes. However, the `@inherits` directive can be used to provide Blazor with a "code-behind" experience that separates view markup from processing code.
+
+The [Components Sample app](https://github.com/aspnet/Docs/tree/master/aspnetcore/client-side/blazor/components/common/samples/) shows how a component can inherit a base class, `BlazorRocksBase`, to provide its properties and methods:
+
+*BlazorRocks.cshtml*:
+
+[!code-cshtml[](common/samples/2.x/ComponentsSample/Pages/BlazorRocks.cshtml?start=1&end=8)]
+
+*BlazorRocksBase.cs*:
+
+[!code-cshtml[](common/samples/2.x/ComponentsSample/Pages/BlazorRocksBase.cs)]
+
+The base class must derive from `BlazorComponent` in the `Microsoft.AspNetCore.Blazor.Components` namespace.
+
+### Render C# expressions
+
+C# expressions are rendered by prefixing the `@` symbol. For example, a C# field is rendered by prefixing `@` to the field name. The following example evaluates and renders:
+
+* The `_HeadingFontStyle` field to the CSS property value for `font-style`.
```

Comment:
The "pascal case with leading underscore" variable name is a very strange naming convention. For normal C# code we go with `camelCase` for private fields (or `_camelCase` for private fields at a push), and `PascalCase` for public properties. No `_PascalCase`.

Reply:
Yes, of course. It was a property that we changed to a field on an earlier commit. I shot the underscore without lower-casing.